oxidation/libparsec/crates/client_types/src/local_device_file.rs

```diff
@@ -80,16 +82,29 @@ pub enum DeviceFile {
 
 impl DeviceFile {
     pub fn save(&self, key_file_path: &Path) -> LocalDeviceResult<()> {
+        use std::fs::Permissions;
+        use std::os::unix::prelude::PermissionsExt;
+
         if let Some(parent) = key_file_path.parent() {
             std::fs::create_dir_all(parent)
                 .map_err(|_| LocalDeviceError::Access(key_file_path.to_path_buf()))?;
+
+            // Set dir perms to 700 (Only user can access it)
+            #[cfg(unix)]
+            std::fs::set_permissions(parent, Permissions::from_mode(0o700))
+                .map_err(|_| LocalDeviceError::Access(key_file_path.to_path_buf()))?;
```

Comment:
Would that change the permission of folder that already exist ?
If `yes` I think we should prevent it.

Or maybe as @touilleMan said, we skip the permission since the critical data in the file are encrypted :thinking:

Reply:
It will overwrite it I think

Reply:
Then it's not something I want :/ (I don't want parsec the change the permission of folder that already exist)

I've found [`DirBuilder`](https://doc.rust-lang.org/std/fs/struct.DirBuilder.html) I would work for our case.

But we can also skip the permission part :thinking:

Reply:
I think I'll skip the permission part, as we said yesterday it adds an unnecessary layer of complexity which is not needed because the files are encrypted.

Reply:
:+1: